File: data/deepseek_coder_base_6.7B/readable_format_HPEC/fortran/file023.f90
program DRB171_input_dependence_var_yes
    use omp_lib
    implicit none

    integer :: i, N, argCount, allocStatus, rdErr, ix
    character(len=80), dimension(:), allocatable :: args
    integer, dimension(:), allocatable :: a

    N = 100

    argCount = command_argument_count()
    if (argCount == 0) then
        write (*,'(a)') "No command line arguments provided."
    end if

    allocate(args(argCount), stat=allocStatus)
    if (allocStatus > 0) then
        write (*,'(a)') "Allocation error, program terminated."
        stop
    end if

    do ix = 1, argCount
        call get_command_argument(ix,args(ix))
    end do

    if (argCount >= 1) then
        read (args(1), '(i10)', iostat=rdErr) N
        if (rdErr /= 0 ) then
            write (*,'(a)') "Error, invalid integer value."
        end if
    end if

    allocate (a(N))

    call load_from_input(a, N)

    
    !$omp parallel do shared(a)
    do i = 1, N
        a(i) = i
        if(N .gt. 10000) a(1) = 1
    end do
    !$omp end parallel do

end program

File: data/llama3_8B_ins_results/readable_format_HPEC/cpp/file023.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <omp.h>

int main(int argc, char* argv[]) {
    int i, N, allocStatus, rdErr, ix;
    std::vector<std::string> args;
    std::vector<int> a;

    N = 100;

    if (argc == 0) {
        std::cout << "No command line arguments provided." << std::endl;
    }

    args.resize(argc);
    if (args.capacity() == 0) {
        std::cout << "Allocation error, program terminated." << std::endl;
        return 1;
    }

    for (ix = 1; ix <= argc; ix++) {
        args[ix - 1] = argv[ix];
    }

    if (argc >= 1) {
        try {
            N = std::stoi(args[0]);
        } catch (std::invalid_argument& e) {
            std::cout << "Error, invalid integer value." << std::endl;
            return 1;
        }
    }

    a.resize(N);

    load_from_input(a, N);

    #pragma omp parallel for shared(a)
    for (i = 1; i <= N; i++) {
        a[i - 1] = i;
        if (N > 10000) a[0] = 1;
    }

    return 0;
}
